app/pages/api/state/index.ts

```diff
@@ -58,10 +76,16 @@ export async function setState(
   );
 
   const branches = await fetchTagsForImage('specify7-service');
-  const state = rawState.map((deployment) => ({
-    ...deployment,
-    digest: branches[deployment.branch]?.digest ?? deployment.digest,
-  }));
+  const state = await Promise.all(
+    rawState.map(async (deployment) => {
+      const hasInteralSp7ConfigDirectory = await branchHasConfigDirectory(deployment.branch);
```

Comment:
this is going to make a lot of network requests to github (setState is called many times per day, and the request is made for each branch)

though I guess this is the least complicated approach